src/components/controls/controls.js

```diff
@@ -0,0 +1,222 @@
+var registerControls = require('../../core/controls').registerControls;
+var THREE = require('../../lib/three');
+
+var MAX_DELTA = 0.2; // ms
+var PI_2 = Math.PI / 2;
+
+/**
+ * Controls component.
+ *
+ * Receives input events from device-specific control components, and applies movement and rotation
+ * to the element accordingly.
+ */
+module.exports.Component = registerControls('controls', {
+
+  dependencies: ['velocity', 'rotation'],
+
+  schema: {
+    enabled: { default: true },
+
+    flyingEnabled: { default: false },
+
+    movement: { default: ['gamepad-controls', 'wasd-controls', 'touch-controls'] },
+    movementEnabled: { default: true },
```

Comment:
I think being able to disable movement seems important. Lots of cardboard demos out now that want to use their one-and-only button for something other than movement, or even for real VR you might want to constrain the camera inside a vehicle, or write your own teleportey-style movement.